core/cmd/remote_client.go

```diff
@@ -548,3 +555,41 @@ func parseResponse(resp *http.Response) ([]byte, error) {
 	}
 	return b, err
 }
+
+func (cli *Client) checkRemoteBuildCompatibility(lggr logger.Logger, c *clipkg.Context) error {
+	resp, err := cli.HTTP.Get("/v2/build_info")
+	if err != nil {
+		lggr.Warn("Remote node version is unknown. CLI may behave in unexpected ways.")
+		return nil
+	}
+	b, err := parseResponse(resp)
+	if err != nil {
+		lggr.Warn("Remote node version is unknown. CLI may behave in unexpected ways.")
+		return nil
+	}
+
+	var remote_build_info map[string]string
+	if err := json.Unmarshal(b, &remote_build_info); err != nil {
+		lggr.Warn("Remote node version is unknown. CLI may behave in unexpected ways.")
+		return nil
+	}
+	remote_version, remote_sha := remote_build_info["version"], remote_build_info["commitSHA"]
+	cli_version, cli_sha := static.Version, static.Sha
+
+	if remote_version == "unset" || cli_version == "unset" {
+		lggr.Warn("Remote node version is unknown. CLI may behave in unexpected ways.")
+		return nil
+	}
```

Comment:
I think we need to consider that either version or sha may be `unset` (or `""` for that matter), and it might be safer to default to a mismatch in that case by expanding the if below instead?

Reply:
From what i understand it shouldn't ever be empty ("") otherwise it will panic here https://github.com/smartcontractkit/chainlink/blob/develop/core/static/static.go#L39-L48 , so i chose to only compare to "unset" here. 

As to whether we should error out here, I was being defensive for unexpected conditions. (Similar to the parseResponse error above) but thinking more I'm fine with erroring out since we now provide the flag to prevent disruption.

Reply:
Hmmm, well we don't have the same check for sha, but it also seems brittle to depend on that implicit condition anyways - particularly since we don't actually know anything about what the remote version will be.

Reply:
yeah, I'll just simplify by treating these cases as mismatch